app/electron/main.ts

```diff
@@ -307,6 +300,32 @@ function setMenu(i18n: I18n) {
   Menu.setApplicationMenu(menu);
 }
 
+async function getRunningHeadlampPIDs() {
+  const processes = await find_process('name', 'headlamp-server.*');
```

Comment:
find_process also works with port. In the future that could be used to detect other not-headlamp-processes squatting on our port.

Reply:
Makes sense. But a fix for that would be to run it with another port or show a warning. We can do that in a different PR.